addons/web_editor/models/ir_attachment.py

```diff
@@ -1,18 +1,53 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, fields
+from werkzeug import url_quote
+
+from odoo import api, models, fields, tools
 
 
 class IrAttachment(models.Model):
 
     _inherit = "ir.attachment"
 
     local_url = fields.Char("Attachment URL", compute='_compute_local_url')
+    image_src = fields.Char(compute='_compute_image_src')
+    image_width = fields.Integer(compute='_compute_image_size')
+    image_height = fields.Integer(compute='_compute_image_size')
 
     @api.one
     def _compute_local_url(self):
         if self.url:
             self.local_url = self.url
         else:
             self.local_url = '/web/image/%s?unique=%s' % (self.id, self.checksum)
+
+    @api.multi
+    @api.depends('mimetype', 'url', 'name', 'datas_fname')
+    def _compute_image_src(self):
+        for attachment in self:
+            if not any(m == attachment.mimetype for m in ['image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/gif', 'image/png', 'image/svg+xml']):
```

Comment:
attachment.mimetype in ['image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/gif', 'image/png', 'image/svg+xml']

Reply:
Oops... changed